types/inquirer/index.d.ts

```diff
@@ -70,90 +71,134 @@ declare namespace inquirer {
         restoreDefaultPrompts(): void;
     }
 
-    interface Question<T = Answers> {
+    interface QuestionCommon<A> {
         /**
-         * Type of the prompt.
-         * Possible values:
-         * <ul>
-         *      <li>input</li>
-         *      <li>number</li>
-         *      <li>confirm</li>
-         *      <li>list</li>
-         *      <li>rawlist</li>
-         *      <li>password</li>
-         * </ul>
-         * @defaults: 'input'
+         * The name to use when storing the answer in the answers hash.
+         * If the name contains periods, it will define a path in the answers hash.
          */
-        type?: string;
+        name?: string;
         /**
-         * The name to use when storing the answer in the anwers hash.
+         * The question to print. If defined as a function, the first parameter will be
+         * the current inquirer session answers.
+         * Defaults to the value of `name` (followed by a colon).
          */
-        name?: string;
+        message?: string | ((answers: A) => string);
+        /**
+         * Default value(s) to use if nothing is entered, or a function that returns
+         * the default value(s). If defined as a function, the first parameter will be
+         * the current inquirer session answers.
+         */
+        default?: string | number | boolean | any[] | ((answers: A) => any) | ((answers: A) => Promise<any>);
+        /**
+         * Change the default _prefix_ message.
+         */
+        prefix?: string;
         /**
-         * The question to print. If defined as a function,
-         * the first parameter will be the current inquirer session answers.
+         * Change the default _suffix_ message.
          */
-        message?: string | ((answers: T) => string);
+        suffix?: string;
         /**
-         * Default value(s) to use if nothing is entered, or a function that returns the default value(s).
-         * If defined as a function, the first parameter will be the current inquirer session answers.
+         * Receive the current user answers hash and should return `true` or `false` depending
+         * on whether or not this question should be asked. The value can also be a simple boolean.
          */
-        default?: any | ((answers: T) => any) | ((answers: T) => Promise<any>);
+        when?: boolean | ((answers: A) => boolean | Promise<boolean>);
+    }
+
+    interface QuestionOptions<A> {
```

Comment:
Although specific type string is defined and given in specific Question interfaces, still not understand why not state `type` property existing in the generic `QuestionCommon` interface

Reply:
The `QuestionCommon` interface is only for props that are shared between ALL question types. This `QuestionOptions` interface is only meant to be picked from, (e.g., `Pick<T, 'choices'>`). The reason I wrote it this way is so that these props only have to be written once, but they can be picked for relevant question types that need them, complete with the TSDoc comments.

Reply:
The props in this interface are used across more than one question type, but not all question types. Does that make sense?